source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -443,6 +456,24 @@ ReadOrParseState Filter::readProxyHeader(Network::ListenerFilterBuffer& buffer)
   } else {
     // continue searching buffer from where we left off
     for (; search_index_ < raw_slice.len_; search_index_++) {
+
+      if (config_.get()->allowRequestsWithoutProxyProtocol()) {
+        if (search_index_ < PROXY_PROTO_V1_SIGNATURE_LEN &&
+            buf[search_index_] != PROXY_PROTO_V1_SIGNATURE[search_index_]) {
+          possibly_v1_ = false;
+        }
+        if (search_index_ < PROXY_PROTO_V2_SIGNATURE_LEN &&
```

Comment:
nit, maybe can check the `!possibly_v1_` first, otherwise skip v2 byte compare.

Reply:
I don't think we can skip v2 byte compare, for the reason outlined in https://github.com/envoyproxy/envoy/pull/18951#discussion_r862910785

I think the only optimization we could make is check if `possibly_v2_` is already true, but I expect this is something the compiler would optimize away anyways